Mastodon/Protocol/Provider/DataSourceProvider+UITableViewDelegate.swift

```diff
@@ -166,30 +175,39 @@ extension UITableViewDelegate where Self: DataSourceProvider & MediaPreviewableV
-                            image: UIImage(systemName: "square.and.arrow.up")!,
-                            identifier: nil,
-                            discoverabilityTitle: nil,
-                            attributes: [],
-                            state: .off
-                        ) { [weak self] _ in
-                            os_log(.info, log: .debug, "%{public}s[%{public}ld], %{public}s: share", ((#file as NSString).lastPathComponent), #line, #function)
-                            guard let self = self else { return }
-                            Task {
-                                let applicationActivities: [UIActivity] = [
-                                    SafariActivity(sceneCoordinator: self.coordinator)
-                                ]
-                                let activityViewController = UIActivityViewController(
-                                    activityItems: [assetURL],
-                                    applicationActivities: applicationActivities
-                                )
-                                activityViewController.popoverPresentationController?.sourceView = mediaView
-                                self.present(activityViewController, animated: true, completion: nil)
-                            }
                         }
-                    ]
+                    )
+                }
+                subMenu.append(
+                    UIAction(
+                        title: L10n.Common.Controls.Actions.share,
+                        image: UIImage(systemName: "square.and.arrow.up")!,
+                        identifier: nil,
+                        discoverabilityTitle: nil,
+                        attributes: [],
+                        state: .off
+                    ) { [weak self] _ in
+                        os_log(.info, log: .debug, "%{public}s[%{public}ld], %{public}s: share", ((#file as NSString).lastPathComponent), #line, #function)
+                        guard let self = self else { return }
+                        Task {
+                            let applicationActivities: [UIActivity] = [
+                                SafariActivity(sceneCoordinator: self.coordinator)
+                            ]
+                            let activityViewController = UIActivityViewController(
+                                activityItems: [assetURL],
+                                applicationActivities: applicationActivities
+                            )
+                            activityViewController.popoverPresentationController?.sourceView = mediaView
+                            self.present(activityViewController, animated: true, completion: nil)
+                        }
+                    }
+                )
+                return UIMenu(
+                    title: "",
+                    image: nil,
+                    identifier: nil,
+                    options: [],
+                    children: subMenu
                 )
```

Comment:
Small nit: this should work too:
```suggestion
                return UIMenu(children: subMenu)
```